### ISSUE TITLE
How to run the semantic segmentation on my own images?

### ISSUE THREAD

**Issue by Tetsujinfr** [itijyou/ademxapp#34]
hi
so I did succesfully run the perf test on Pascal VOC with:
`python` issegm/voc.py --data-root data/VOCdevkit --output output --phase val --weights models/voc_rna-a1_cls21_s8_ep-0001.params --split val --test-scales 500 --test-flipping --gpus 0`

Now I only want to use the pre-trained model and test it on non-VOC images, so on my own images, but it seems that voc.py is highly specific to the voc dataset.

Can I use voc.py and some flag to run it on my own images or do I need to heavily modify the code so that it does not look for segmentationClass mask images from voc etc?

thanks
Tets

**Comment by Tetsujinfr** [comment]
ok so I managed to run it on my own images but I have some large staircase type of effects, not sure why? is it due to the pre-trained models which has run on low res images and consequently if I run on high resolution images some interpolation is done to upscale the segmentation? Do I need to tweak the code to get better results or this is a model limitation ? (the below is a person standing and his legs do not have such a staircase profil)
![out-0416](https://user-images.githubusercontent.com/29961693/29244053-d6e4f72e-7fa6-11e7-9414-f81d3f26cc55.png)

**Comment by gnobitab** [comment]
Hi, Tetsujinfr. Thanks for your share. May I ask how I can get a segmented image on my own image like the picture you posted?

**Comment by gnobitab** [comment]
@Tetsujinfr

**Comment by Tetsujinfr** [comment]
Hi sure. Let me post the bits and pieces I did modify later today (can not right now). Warning: this is a quick and dirty hack of the code bit it works to test on new data. Will post the code and and an example of th command line.

**Comment by gnobitab** [comment]
Thanks a lot.

**Comment by Tetsujinfr** [comment]
ok, I attach the modified file data.py and the modified voc.py (voctets.py here). Basically I de-acivated the part which is doing the validation scoring in the voc file. So put your images into a folder, say ademxapp/myInputImages, then go into this folcer and do a "ls *.jpg 
> myInputImages.lst" to generate the file containing the images inventory to process. Then call voctets.py as sudo with --data-root /myInputImages and it should work.
An example:
`python issegm/tetsvoc.py --data-root data/myPNGImages --output outputTets --phase val --weights models/voc_rna-a1_cls21_s8_ep-0001.params --split val --test-scales 720 --test-flipping --gpus 0`

(I did attach the py files as .txt so that I can attach them but please remove the .txt extension)
Hope it helps.

I did run the model over a few video, have a look at my youtube channel, I will post some results soon:
https://www.youtube.com/channel/UCZUuytmk4SbmKVDgqhq24Kw
Some hint:
![img_5425](https://user-images.githubusercontent.com/29961693/29344430-b8afe710-822f-11e7-91cc-5ccbc39f236e.png)
![img_8793](https://user-images.githubusercontent.com/29961693/29344633-2293678c-8231-11e7-8abf-e754aa8a3293.png)

Tets

[data.py.txt](https://github.com/itijyou/ademxapp/files/1226861/data.py.txt)
[tetsvoc.py.txt](https://github.com/itijyou/ademxapp/files/1226862/tetsvoc.py.txt)

**Comment by gnobitab** [comment]
Thanks for your share. Great help.

**Comment by Tetsujinfr** [comment]
I will close this issue in the next couple days if no more question.

**Comment by jeremy-cv** [comment]
Hi Testsujinfr,

thanks for your instructions. Did you finally find out why there is this "staircase effect" ?

Thank you

**Comment by Tetsujinfr** [comment]
Not fully although I suspect it is due to the resolution of the training dataset of the model, which I think take 224x224 pixel input images. So ultimately for large hi res image, the model is going to interpolate somehow, and it seems it is doing that is a squarry way as a default. I did not dig it further. I liked this project, one of the best results I have seen on segmentation so far, but I am not an expert in the field.

**Comment by johnnylu305** [comment]
Although the issue has been closed, if you want to train/test the network with a personal dataset, please edit the "train.lst/val.lst" in the "issegm/data/voc/" and replace the images in the "data/VOCdevkit/VOC2012/JPEGImages" and "data/VOCdevkit/VOC2012/SegmentationClass" with yours.
.
